core/services/ardupilot_manager/mavlink_proxy/test_all.py

```diff
@@ -13,17 +15,32 @@
 
 
 def test_endpoint() -> None:
-    endpoint = Endpoint("udp:0.0.0.0:14550")
-    assert endpoint.connType == EndpointType.UDPClient, "Connection type does not match."
-    assert endpoint.place == "0.0.0.0", "Connection place does not match."
-    assert endpoint.argument == "14550", "Connection argument does not match."
-    assert endpoint.__str__() == "udp:0.0.0.0:14550", "Connection string does not match."
-
-    endpoint = Endpoint({"connType": "udp", "place": "0.0.0.0", "argument": "14550"})
-    assert endpoint.connType == EndpointType.UDPClient, "Connection type does not match."
+    endpoint = Endpoint("udpout", "0.0.0.0", 14550)
+    assert endpoint.connection_type == EndpointType.UDPClient, "Connection type does not match."
     assert endpoint.place == "0.0.0.0", "Connection place does not match."
-    assert endpoint.argument == "14550", "Connection argument does not match."
-    assert endpoint.__str__() == "udp:0.0.0.0:14550", "Connection string does not match."
+    assert endpoint.argument == 14550, "Connection argument does not match."
+    assert endpoint.__str__() == "udpout:0.0.0.0:14550", "Connection string does not match."
+
+
+def test_endpoint_validators() -> None:
+    endpoint_values = {"connection_type": "udpin", "place": "0.0.0.0", "argument": -30}
+    with pytest.raises(ValueError):
+        Endpoint.is_mavlink_endpoint(endpoint_values)
+    endpoint_values = {"connection_type": "udpin", "place": "42", "argument": 14555}
+    with pytest.raises(ValueError):
+        Endpoint.is_mavlink_endpoint(endpoint_values)
+    endpoint_values = {"connection_type": "serial", "place": "dev/autopilot", "argument": 115200}
+    with pytest.raises(ValueError):
+        Endpoint.is_mavlink_endpoint(endpoint_values)
+    endpoint_values = {"connection_type": "file", "place": "/path/to/file", "argument": -30}
+    with pytest.raises(ValueError):
+        Endpoint.is_mavlink_endpoint(endpoint_values)
+    endpoint_values = {"connection_type": "file", "place": "path/to/file", "argument": 100}
+    with pytest.raises(ValueError):
+        Endpoint.is_mavlink_endpoint(endpoint_values)
+    endpoint_values = {"connection_type": "potato", "place": "path/to/file", "argument": 100}
+    with pytest.raises(ValueError):
+        Endpoint.is_mavlink_endpoint(endpoint_values)
```

Comment:
Can you add tests for valid endpoint configuration ?

Reply:
Sure.